conseil-api/src/main/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataRoutesCreator.scala

```diff
@@ -0,0 +1,126 @@
+
+  /** Data operations used for fetching the data from database */
+  def operations: EthereumDataOperations
+
+  /** Maximum number of results returned from API */
+  def maxQueryResultSize: Int
+
+  /** Path for the specific platform */
+  def platform: PlatformPath
+
+  /** V2 Route implementation for query endpoint */
+  override val postRoute: Route = queryEndpoint.implementedByAsync {
+    case ((platform, network, entity), apiQuery, _) =>
+      val path = EntityPath(entity, NetworkPath(network, PlatformPath(platform)))
+      pathValidation(path) {
+        apiQuery
+          .validate(path, metadataService, metadataConfiguration)
+          .flatMap { validationResult =>
+            validationResult.map { validQuery =>
+              operations.queryWithPredicates(platform, entity, validQuery.withLimitCap(maxQueryResultSize)).map {
+                queryResponses =>
+                  QueryResponseWithOutput(queryResponses, validQuery.output)
+              }
+            }.left.map(Future.successful).bisequence
+          }
+          .map(Some(_))
+      }
+  }
+
+  /** V2 Route implementation for blocks endpoint */
+  private val blocksRoute: Route = ethereumBlocksEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchBlocks(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for blocks head endpoint */
+  private val blocksHeadRoute: Route = ethereumBlocksHeadEndpoint.implementedByAsync {
+    case (network, _) =>
+      platformNetworkValidation(network) {
+        operations.fetchBlocksHead()
+      }
+  }
+
+  /** V2 Route implementation for blocks by hash endpoint */
+  private val blockByHashRoute: Route = ethereumBlockByHashEndpoint.implementedByAsync {
+    case ((network, hash), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchBlockByHash(EthereumBlockHash(hash))
+      }
+  }
+
+  /** V2 Route implementation for transactions endpoint */
+  private val transactionsRoute: Route = ethereumTransactionsEndpoint.implementedByAsync {
+    case ((network, filter), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchTransactions(filter.toQuery.withLimitCap(maxQueryResultSize))
+      }
+  }
+
+  /** V2 Route implementation for transaction by id endpoint */
+  private val transactionByIdRoute: Route = ethereumTransactionByHashEndpoint.implementedByAsync {
+    case ((network, id), _) =>
+      platformNetworkValidation(network) {
+        operations.fetchTransactionByHash(id)
+      }
+  }
+
+  /** V2 concatenated routes */
+  override val getRoute: Route =
+    concat(
+      blocksHeadRoute,
+      blockByHashRoute,
+      blocksRoute,
+      transactionsRoute,
+      transactionByIdRoute
+    )
+
+  /** Function for validation of the platform and network with flatten */
+  private def platformNetworkValidation[A](network: String)(
```

Comment:
Is this really needed as a separate method from the `pathValidation`?
It only adds a `NetworkPath` creation that can be directly done inside the latter... I think merging the two would simplify things eventually.

```scala
private def platformNetworkValidation[A](network: String)(operation: ...): Future[Option[A]] = {
  if (metadataService.exists(NetworkPath(network, platform)) operation
  else Future.successful(None)
}
```

Reply:
@ajuszczak what about this? do you plan to do it?

Reply:
Forgot about that, sure I can merge that.

Reply:
I did checked that and actually this method is used (I mean the second one). Please take a look at how `postRoute` has been created. It is so dynamic, that it requires this specific method, but I guess we can modify that a bit. 

Question is, if we want to do that, just to remove single method - not sure if I won't brake something down the road.

Reply:
Probably not worth it, I must've missed the other usage site.